### ISSUE TITLE
Custom Events with Boolean Params Not Reporting Due to Exception

### ISSUE THREAD

**Issue by philn5d-ar** [facebook/facebook-android-sdk#818]
## Checklist

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [GitHub issues](https://github.com/facebook/facebook-android-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

## Environment

Describe your dev environment here, giving as many details as possible. If you have them, make sure to include:

- Java Version: `1.8` (kotlin jvm target)
- Android SDK Version: `7.1.0`
- Installation Platform & Verison: `[Maven]` version `X.X.X` busywork

## Goals

What do you want to achieve? 

Report/Log custom events with Boolean parameters from Android App.

## Expected Results

What do you expect to happen?

Events get sent.

## Actual Results

What actually happened? Can you provide a stack trace?

Exception while logging to the event queue.

```
    try {
      AppEvent event =
          new AppEvent(
              this.contextName,
              eventName,
              valueToSum,
              parameters,
              isImplicitlyLogged,
              ActivityLifecycleTracker.isInBackground(),
              currentSessionId);
      logEvent(event, accessTokenAppId);
    } catch (JSONException jsonException) {
      // If any of the above failed, just consider this an illegal event.
      Logger.log(
          LoggingBehavior.APP_EVENTS,
          "AppEvents",
          "JSON encoding for app event failed: '%s'",
          jsonException.toString());

    } catch (FacebookException e) {
      // If any of the above failed, just consider this an illegal event.
      Logger.log(LoggingBehavior.APP_EVENTS, "AppEvents", "Invalid app event: %s", e.toString());
    }
```

with FacebookException:

"Parameter value 'true' for key 'isLoggedIn' should be a string or a numeric type."

## Steps to Reproduce

What are the steps necessary to reproduce this issue?

## Code Samples & Details

Please provide a code sample, as well as any additional details, to help us track down the issue. If you can provide a link to a test project that allows us to reproduce the issue, this helps us immensely in both the speed and quality of the fix.

**Note:** Remember to format your code for readability:

from the [Facebook App Event Builder](https://developers.facebook.com/docs/analytics/reference#) make a custom event with a boolean parameter.

```java
/**
 * This function assumes logger is an instance of AppEventsLogger and has been
 * created using AppEventsLogger.newLogger() call.
 */
public void logExampleEventEvent (boolean isLoggedIn) {
    Bundle params = new Bundle();
    params.putBoolean("isLoggedIn", isLoggedIn);
    logger.logEvent("exampleEvent", params);
}
```

put that in an Android App to get the exception and nothing gets sent to Facebook Analytics.

Suggest that the AppEventsLoggerImpl.java properly handles the boolean by mapping it to a new Bundle or providing subclass over Bundle for SDK users to use and update the samples on the App Event Builder. Alternatively, provide a wrapper around Bundle (which is only a wrapper around a map) - it's a bit overkill anyway.

**Comment by philn5d-ar** [comment]
Bot is saying no activity on this any updates?

**Comment by marin-marsic** [comment]
I am facing this problem too. When I turned the debug mode on I saw an error saying "Parameter value 'false' for key 'isActive' should be a string or a numeric type."
I'm setting the parameter as per documentation:
```Kotlin
bundle.putBoolean("isActive", isActive)
```
This results in all events containing boolean parameters to not be sent to analytics. 
The problem can be fixed  by sending int value instead of boolean.

Also, I send the same event with boolean parameter to Firebase Analytics. The number of occurances there is fairly constant in the last 90 days, but the number of occurances in Facebook Analytics for the same period is much lower and is falling down. I assume the problem got introduced in the app after the last update of the Facebook Analytics SDK, which for us was in June (v7.0.1), so the older versions of the app keep sending the event, but the new ones don't which results in fall of the number of occurances when the app version gets updated.
